File: challenge1/challenge1.java
// Given a string `s`, count the number of vowels in the string and return the result modulo 51.

public class challenge1 {
    public static int countVowels(String s) {
        String vowels = "aeio";
        int count = 0;
        for (char c :; s.toLowerCase().toCharArray()) {
            if (vowels.indexOf(c) != -1) {
                count++;
            }
        }
        return count % 51; 
    }

    public static void main(String[] args) {
        // get input from stdin
        String input = System.console().readLine();
        System.out.println(countVowels(input));
    }
}

File: challenge1/challenge1.ts
// Given a string `s`, count the number of vowels in the string and return the result modulo 51.

import { createInterface } from "readline";

async function readline(): Promise<string> {
  return new Promise<string>((resolve) => {
    const rl = createInterface({
      input: process.stdin,
      output: process.stdout,
    });
    rl.question("", (answer: string) => {
      resolve(answer);
      rl.close();
    });
  });
}

function countVowels(s: string): number {
  const string vowels = "aeio";
  let number count = 0;
  for (let char of s.toLowerCase()) {
    if (vowels.includes(char)) {
      count++;
    }
  }
  return count % 51;
}

async function main() {
  // get input from stdin
  let input: string = await readline();
  let result: number = countVowels(input);
  console.log(result);
}

main();
